src/BuildExecutable.jl

```diff
@@ -252,22 +252,40 @@ function emit_cmain(cfile, exename, relocation)
             {
                 #if defined(_WIN32) || defined(_WIN64)
                 char *julia_home_env = (char*)malloc(12 + strlen(julia_home));
-                sprintf(julia_home_env, \"JULIA_HOME=%s\", julia_home);
+                sprintf(julia_home_env, "JULIA_HOME=%s", julia_home);
                 assert(_putenv(julia_home_env) == 0);
                 free(julia_home_env);
                 #else
-                assert(setenv(\"JULIA_HOME\", julia_home, 1) == 0);
+                assert(setenv("JULIA_HOME", julia_home, 1) == 0);
                 #endif
                 free(julia_home);
             }
 
+            // set Base.ARGS, not Core.ARGS
```

Comment:
I tried something very similar to this earlier to day, but it did not succeed, probably because I tried to set `Core.ARGS`:).